File: dispense_app_ang/src/app/model/ue-full-dto.ts
/**
 * DTO complet d'une UE retourné par GET /api/ue/detail/{code},
 * incluant les acquis d'apprentissage.
 *
 * @author Ludovic
 */

export interface AcquisDto {
  num: number;
  acquis: string;
  pourcentage: number;
}

export interface UeFullDto {
  code: string;
  ref: string;
  nom: string;
  nbPeriodes: number;
  ects: number;
  prgm: string;
  acquis: AcquisDto[];
}

File: src/main/java/org/isfce/pid/dto/UEFullDto.java
package org.isfce.pid.dto;

import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;

/**
 * DTO complet d'une UE avec ses acquis.
 * @author Ludovic
 */
public record UEFullDto(
	@NotBlank String code,
	@NotBlank String ref,
	@NotBlank String nom,
	@Min(value = 1, message = "{err.ue.nbPeriodes}") int nbPeriodes,
	@Min(value = 1, message = "{err.ue.nbECTS}") int ects,
	@NotBlank String prgm,
	@Size(min = 1, message = "{err.ue.acquis}") @NotEmpty @Valid List<AcquisFullDto> acquis
) {}
